pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -225,6 +225,7 @@ func readConfig(config io.Reader) (VSphereConfig, error) {
 }
 
 func init() {
+	registerMetrics()
```

Comment:
Prometheus metrics is a kind of add on. Should we stop with the cloud provider initialization, if this fails? 
I see registerMetrics() panics if it fails to register the vSphere metrics -vsphereApiMetric, vsphereApiErrorMetric.

Instead of using MustRegister(), you can use Register() to register the vSphere metrics and log an error if possible.

We can discuss whether we want to panic or goahead with CP initialization.